File: frontend/lib/types/product.ts
import { AuditFieldsCreate, AuditFieldsUpdate, EntityAuditFields } from "./common";

export interface Product extends EntityAuditFields {
  id: string;
  external_ref: string | null;
  sku: string;
  name: string;
  description: string | null;
  uom: string | null;
  price: number;
  currency: string;
  tax_code: string | null;
  is_active: boolean;
}

export interface ProductCreate extends AuditFieldsCreate {
  external_ref?: string;
  sku: string;
  name: string;
  description?: string;
  uom?: string;
  price: number;
  currency?: string;
  tax_code?: string;
  is_active?: boolean;
}

export interface ProductUpdate extends AuditFieldsUpdate {
  external_ref?: string;
  sku?: string;
  name?: string;
  description?: string;
  uom?: string;
  price?: number;
  currency?: string;
  tax_code?: string;
  is_active?: boolean;
}

File: backend/app/schemas/product.py
import uuid
from pydantic import BaseModel, Field


class ProductBase(BaseModel):
    external_ref: str | None = None
    sku: str
    name: str
    description: str | None = None
    uom: str | None = None
    price: float
    tax_code: str | None = None
    is_active: bool = True


class ProductCreate(ProductBase):
    pass


class ProductUpdate(BaseModel):
    external_ref: str | None = None
    name: str | None = None
    description: str | None = None
    uom: str | None = None
    price: float | None = None
    tax_code: str | None = None
    is_active: bool | None = None


class ProductOut(ProductBase):
    id: uuid.UUID

    class Config:
        orm_mode = True

